recipe/meta.yaml

```diff
@@ -1,24 +1,27 @@
-{% set version = "2.0.0" %}
+{% set version = "3.1.0" %}
+# Triton no longer tags releases.  We're taking the commit id from
+# PyTorch's .ci/docker/ci_commit_pins/triton.txt 
+{% set git_commit = "5fe38ffd73c2ac6ed6323b554205186696631c6f" %}
 
 package:
   name: triton
   version: {{ version }}
 
 source:
-  url: https://github.com/openai/triton/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: 19b6de0d0bcce86e973258d112169cd321b677146808bcc5fed7f69046775cfd
+  url: https://github.com/openai/triton/archive/{{ git_commit }}.tar.gz
+  sha256: 933babc32b69872efbce05fe8be61129fecf52c724fadea42d8c7b2d10e16ad9
   patches:
-    - patches/0001-do-not-package-third_party-folder.patch
-    - patches/0002-Fix-error-that-FileCheck-is-already-defined.patch
-    - patches/0003-properly-point-to-triton-includes.patch
-    - patches/0004-Avoid-using-outdated-FindLLVM.patch
-    - patches/0005-Unvendor-third-party-libs.patch
-    - patches/0006-Avoid-pip-check-error-as-cmake-is-not-a-python-modul.patch
-    - patches/0007-Fix-TableGen-issues.patch
-    - patches/0008-Search-for-libs-in-CONDA_PREFIX-instead-of-third_par.patch
```

Comment:
It seems triton changed a lot since v2.0. A casual check against [`python/setup.py`](https://github.com/triton-lang/triton/blob/main/python/setup.py) shows that indeed most of that doesn't apply anymore.

Reply:
I've checked 1..7 in detail, and for 8 I've assumed it's unnecessary given it seems to find the libraries just fine at runtime (the patched code is gone).

Reply:
Yeah, I'm not against dropping the patches, I was just wondering if they _all_ ended up useless/obsolete. This seems to be the case, so that's great!